R/developer.R

```diff
@@ -0,0 +1,162 @@
+#' @name developer_functions
+#'
+#' @title Developer functions for creating recipes steps
+#'
+#' @description
+#'
+#' This page provides a comprehensive list of the exported functions for
+#' creating recipes steps, and how to use them.
+#'
+#' @details
+#'
+#' # Creating steps
+#'
+#' [add_step()] and [add_check()] are required when creating a new step. It
+#' should be the other function call in your steps and should have the following
+#' format:
+#'
+#' ```r
+#' step_example <- function(recipe,
+#'                          ...,
+#'                          role = NA,
+#'                          trained = FALSE,
+#'                          skip = FALSE,
+#'                          id = rand_id("example")) {
+#'   add_step(
+#'     recipe,
+#'     step_example_new(
+#'       terms = enquos(...),
+#'       role = role,
+#'       trained = trained,
+#'       skip = skip,
+#'       id = id
+#'     )
+#'   )
+#' }
+#' ```
+#'
+#' [rand_id()] should be used in the arguments of `step_example()` to specify
+#' the argument, as we see in the above example.
+#'
+#' [recipes_pkg_check()] should be used in `step_example()` functions together
+#' with [required_pkgs()] to alert users that certain other packages are
+#' required. The standard way of using this function is the following format:
+#'
+#' ```r
+#' recipes_pkg_check(required_pkgs.step_example())
```

Comment:
Is it actually standard to write out the name of the method? If not, I'd write it using only the name of the `required_pkgs()` generic and note that it ought to dispatch to the `step_example` method.

Reply:
This is actually how we use it! 😄 

https://github.com/tidymodels/recipes/blob/66a1542b861af752703eb5b4593e71235df05622/R/pls.R#L143

we have to do it this way since we call it before an `step_pls` object is created for `required_pkgs()` to dispatch off

Reply:
Okay, gotcha! Thumbs up, then. :)